File: modules/angular2/di.ts
/**
 * @module
 * @description
 * The `di` module provides dependency injection container services.
 */

export {
  InjectMetadata,
  OptionalMetadata,
  InjectableMetadata,
  SelfMetadata,
  HostMetadata,
  SkipSelfMetadata,
  DependencyMetadata
} from './src/core/di/metadata';

// we have to reexport * because Dart and TS export two different sets of types
export * from './src/core/di/decorators';

export {forwardRef, resolveForwardRef, ForwardRefFn} from './src/core/di/forward_ref';
export {
  Injector,
  ProtoInjector,
  BindingWithVisibility,
  DependencyProvider,
  Visibility,
  UNDEFINED
} from './src/core/di/injector';
export {Binding, BindingBuilder, ResolvedBinding, Dependency, bind} from './src/core/di/binding';
export {Key, KeyRegistry, TypeLiteral} from './src/core/di/key';
export {
  NoBindingError,
  AbstractBindingError,
  CyclicDependencyError,
  InstantiationError,
  InvalidBindingError,
  NoAnnotationError,
  OutOfBoundsError
} from './src/core/di/exceptions';
export {OpaqueToken} from './src/core/di/opaque_token';

File: lib/src/core/di.dart
/**
 * @module
 * @description
 * The `di` module provides dependency injection container services.
 */
library angular2.src.core.di;

export "di/metadata.dart"
    show
        InjectMetadata,
        OptionalMetadata,
        InjectableMetadata,
        SelfMetadata,
        HostMetadata,
        SkipSelfMetadata,
        DependencyMetadata;
// we have to reexport * because Dart and TS export two different sets of types
export "di/decorators.dart";
export "di/forward_ref.dart" show forwardRef, resolveForwardRef, ForwardRefFn;
export "di/injector.dart" show Injector;
export "di/provider.dart"
    show
        Binding,
        ProviderBuilder,
        ResolvedBinding,
        ResolvedFactory,
        Dependency,
        bind,
        Provider,
        ResolvedProvider,
        provide;
export "di/key.dart" show Key, TypeLiteral;
export "di/exceptions.dart"
    show
        NoProviderError,
        AbstractProviderError,
        CyclicDependencyError,
        InstantiationError,
        InvalidProviderError,
        NoAnnotationError,
        OutOfBoundsError;
export "di/opaque_token.dart" show OpaqueToken;
